src/benchq/problem_ingestion/molecule_instance_generation.py

```diff
@@ -14,58 +18,61 @@
-    active space. Note that the active space can be specified in one of the following
-    ways:
+    # make sure wave function is stable before we proceed
+    mean_field_object = stability(mean_field_object)
 
-    1. the use of Atomic Valence Active Space (AVAS),
+    # localize before automatically selecting active space with AVAS
+    mean_field_object = localize(
+        mean_field_object, loc_type="pm"
+    )  # default is loc_type ='pm' (Pipek-Mezey)
 
-    2. by specifying the indices of the occupied and active orbitals,
+    return avas_active_space(mean_field_object, ao_list=ao_list, minao=minao)
 
-        If freeze_core option is True, chemical frozen core orbitals
-        are choosen for occuped_indicies.
 
-    3. by using the frozen natural orbital (FNO) approach.
-       In this case, a user needs to set one of the FNO parameters:
-            - fno_percentage_occupation_number
-            - fno_threshold
-            - fno_n_virtual_natural_orbitals
-        to decide how much virtual space will be kept for the active space.
+def _create_mlflow_setup(
+    mlflow_experiment_name: str, orq_workspace_id: str
+) -> Tuple[MlflowClient, str]:
+    client = MlflowClient(
+        tracking_uri=sdk.mlflow.get_tracking_uri(workspace_id=orq_workspace_id)
+    )
 
-        If freeze_core option is True, chemical frozen core orbitals
-        are choosen for occuped_indicies.
+    experiment = client.get_experiment_by_name(name=mlflow_experiment_name)
+    if experiment is None:
+        client.create_experiment(mlflow_experiment_name)
+        experiment = client.get_experiment_by_name(name=mlflow_experiment_name)
 
+    run = client.create_run(experiment.experiment_id)
+
+    return client, run.info.run_id
+
+
+class SCFConvergenceError(Exception):
+    pass
 
-    Args:
-        geometry: A list of tuples of the form (atom, (x, y, z)) where atom is the
-            atom type and (x, y, z) are the coordinates of the atom in Angstroms.
-        basis: The basis set to use for the calculation.
-        multiplicity: The spin multiplicity of the molecule.
-        charge: The charge of the molecule.
-        avas_atomic_orbitals: A list of atomic orbitals to use for (AVAS).
-        avas_minao: The minimum active orbital to use for AVAS.
-        occupied_indices: A list of molecular orbitals not in the active space that
-            should be assumed to be fully occupied.
-        active_indices: A list of molecular orbitals to include in the active space.
-        freeze_core: A boolean specifying whether frozen core orbitals are selected
-                    for calculations.
-        fno_percentage_occupation_number: Percentage of total occupation number.
-        fno_threshold: Threshold on NO occupation numbers.
-        fno_n_virtual_natural_orbitals: Number of virtual NOs to keep.
-    """
 
+@dataclass(frozen=True)
+class ChemistryApplicationData:
```

Comment:
My inclination would be to preserve the public API of `ChemistryApplicationInstance` in this pull request. Although there are good reasons to change it, it would probably require some investigation to figure out the right way to do it and I wouldn't want to hold up the features added in this PR.